doc/src/modules/physics/mechanics/examples/four_bar_linkage_example.rst

```diff
@@ -48,30 +56,28 @@ Next, we also define the first three joints. ::
    >>> joint3 = PinJoint('J3', link3, link4, coordinates=q3, speeds=u3,
    ...                   parent_point=l3 / 2 * link3.x,
    ...                   child_point=-l4 / 2 * link4.x, joint_axis=link3.z)
+   >>> system.add_joints(joint1, joint2, joint3)
 
 Now we can formulate the holonomic constraint that will close the kinematic
 loop. ::
 
-   >>> loop = link4.masscenter.pos_from(link1.masscenter) + l1 / 2 * link1.x + l4 / 2 * link4.x
-   >>> fh = Matrix([loop.dot(link1.x), loop.dot(link1.y)])
+   >>> loop = (link4.masscenter.pos_from(link1.masscenter) +
+   ...         l1 / 2 * link1.x +l4 / 2 * link4.x)
+   >>> system.add_holonomic_constraints(loop.dot(link1.x), loop.dot(link1.y))
 
-In order to generate the equations of motions, we will use the ``JointsMethod``
-as our fronted. Before setting up the :class:`~.KanesMethod` as its backend we
-need to calculate the velocity constraints. ::
+Before generating the equations of motion we need to specify which generalized
+coordinates and speeds are independent and which are dependent. After which we
+can run :meth:`~.System.validate_system` to do some basic consistency checks. ::
 
-   >>> method = JointsMethod(link1, joint1, joint2, joint3)
-   >>> t = dynamicsymbols._t
-   >>> qdots = solve(method.kdes, [q1.diff(t), q2.diff(t), q3.diff(t)])
-   >>> fhd = fh.diff(t).subs(qdots)
+   >>> system.q_ind = [q1]
+   >>> system.u_ind = [u1]
+   >>> system.q_dep = [q2, q3]
+   >>> system.u_dep = [u2, u3]
+   >>> system.validate_system()
```

Comment:
I would have expected this to return something, does it not return True, for example?

Reply:
I wrote this before my final comment, so maybe we consider that first.